astroquery/esa/xmm_newton/core.py

```diff
@@ -94,48 +102,44 @@ def download_data(self, observation_id, *, filename=None, verbose=False,
             file format, optional, by default all formats
             values: ASC, ASZ, FTZ, HTM, IND, PDF, PNG
 
-
         Returns
         -------
         None if not verbose. It downloads the observation indicated
         If verbose returns the filename
         """
-        if filename is not None:
-            filename = os.path.splitext(filename)[0]
 
-        link = self.data_aio_url + "obsno=" + observation_id
+        # create url to access the aio
+        link = self._create_link(observation_id, **kwargs)
 
-        link = link + "".join("&{0}={1}".format(key, val)
-                              for key, val in kwargs.items())
+        # If the user wants to access proprietary data, ask them for there credentials
+        if prop:
+            username, password = self._get_username_and_password(credentials_file)
+            link = f"{link}&AIOUSER={username}&AIOPWD={password}"
 
         if verbose:
             log.info(link)
 
-        # we can cache this HEAD request - the _download_file one will check
-        # the file size and will never cache
-        response = self._request('HEAD', link, save=False, cache=cache)
-
-        # Get original extension
-        if 'Content-Type' in response.headers and 'text' not in response.headers['Content-Type']:
-            _, params = cgi.parse_header(response.headers['Content-Disposition'])
-        else:
-            if response.status_code == 401:
-                error = "Data protected by proprietary rights. Please check your credentials"
-                raise LoginError(error)
-        response.raise_for_status()
-
+        # get response of created url
+        params = self._request_link(link, cache)
         r_filename = params["filename"]
         suffixes = Path(r_filename).suffixes
 
-        if filename is None:
-            filename = observation_id
-
-        filename += "".join(suffixes)
-
-        self._download_file(link, filename, head_safe=True, cache=cache)
+        # get desired filename
+        filename = self._create_filename(filename, observation_id, suffixes)
+        """
+        If prop we change the log level so that it is above 20, this is to stop a log.debug (line 431) in query.py.
+        This debug reveals the url being sent which in turn reveals the users username and password
+        """
```

Comment:
Can we instead avoid putting the password into the URL?  `log.debug` is needed for actual debugging in many cases; disabling it here is potentially going to hide errors we need to see in the future.  I'm not comfortable overriding it like this.

Reply:
For example, `_download_file` supports the `auth` keyword:
https://github.com/astropy/astroquery/blob/main/astroquery/query.py#L324
so we could do `self._download_file(link_with_no_password, auth={'username': username, 'password': password})`

Reply:
I have tried to use the _download_file auth keyword however the only way for the aio server to download the proprietary data is by having the username and password attached to the end of the url. The _download_file which in turn leads to the requests method does not attach the auth keyword to the end of the url so we cannot use this functionality.

Reply:
OK, I see.  If that's all the server supports, then we're stuck with this approach.